### ISSUE TITLE
Proposal: data mining on positions of users for the 2 repos

### ISSUE THREAD

**Issue by KOLANICH** [rms-support-letter/rms-support-letter.github.io#1474]
*For each repo of {`rms-support-letter/rms-support-letter.github.io`, `rms-open-letter/rms-open-letter.github.io`}:
    * get via GH API nicknames of {`stargazers`, `forks` `owner`s}

* for each user compute a vector describing its membership in orgs and companies (1 - member, 0 - non-member). Can be detected by orgs and by company field in user's profile

* compute Euler's diagram (a - b, b - a, a ^ b)
* for nonintersecting users compute correlation of their position to the company they are member of
* sort the results and plot the nice plots

It also may be possible to train an XGBoost model, and then apply SHAP.

**Comment by 13sqrt5** [comment]
That's highly unethical, i think.

**Comment by KOLANICH** [comment]
What is unethical? Analysing corporate culture based on publicly available information people themselves willfully made publicly available ?

**Comment by 13sqrt5** [comment]
Well, yes. You ask people to sign a letter to support Stallman, and then use their personal data for data mining. That's what corporations are doing, and what they are hated for: collecting personal data and and then selling/abusing it.

**Comment by pizdjuk** [comment]
I support @13sqrt5  hear. Meanwhile finding this idea meaningful. So, @KOLANICH, it would be interesting to make this stuff. But   discuss it elsewhere, and private. Without  publicity.

**Comment by 13sqrt5** [comment]
But indeed, the idea is cool. I think that's an interesting research, and you should make it, but some time later, after the situation with Stallman is resolved.

**Comment by Shamar** [comment]
A more interesting study that I think would have a huge value is to compute the percentage of signers that received money or worked on projects that received money from GAFAM&friends (e.g. GSoC, sponsorship and so on).

This is not something you can automate, but I think that once the signatures will settle it could have a huge sociological (and journalistic) value.

**Comment by shenlebantongying** [comment]
![image](https://user-images.githubusercontent.com/20123683/112417848-c2a35d00-8cfe-11eb-884a-cc9991df4bef.png)
![image](https://user-images.githubusercontent.com/20123683/112417863-cafb9800-8cfe-11eb-8951-f4d9bbd3f389.png)

Hahaha, they are really blind. They even include some signs from Microsoft...........

**Comment by imachug** [comment]
We could probably collect anonymous-ish data, like the percentage of corporate users here vs there and such, though not names and don't build a model. Some simple graphs would be nice to look at and wouldn't harm anyone's privacy.

**Comment by nukeop** [comment]
Organizations:

    FSF: 69
    Open Source Initiative: 8
    EFF: 6
    Collabora: 45
    GitHub: 14
    Gitlab: 1
    Mozilla: 24
    Tor: 8
    Google: 3
    Microsoft: 2
    Red Hat: 8

    Distro:

    Debian: 98
    Fedora: 39
    Ubuntu: 23
    Arch: 23
    OpenSUSE: 11
    Gentoo: 3
    Void: 1
    OpenBSD: 1
    FreeBSD: 1

    DE/WM:

    GNOME: 60
    KDE: 13
    xfce: 0
    cinnamon: 1
    i3/dwm/sway: 0

    Others:

    Guix: 4
    Libreoffice: 3
    Krita: 4

**Comment by imachug** [comment]
Is that from our repo or their repo?

**Comment by shenlebantongying** [comment]
Debian is holding a vote on general solution on the letter.

OMG, Debian is corrupted by a small group of people.

[General resolution: ratify https://github.com/rms-open-letter/rms-open-letter.github.io](https://lists.debian.org/debian-vote/2021/03/msg00083.html)
[Willingness to share a position statement?](https://lists.debian.org/debian-vote/2021/03/msg00061.html)

**Comment by KOLANICH** [comment]
>We could probably collect anonymous-ish data, like the percentage of corporate users here vs there and such, though not names and don't build a model. Some simple graphs would be nice to look at and wouldn't harm anyone's privacy.

To collect "anonymous-ish" data you need to collect non-anonymous one first. And I see no issue in processing non-anonymous data: it is already public data

**Comment by imachug** [comment]
My point is don't list the names of the supporters to blame or something like that. This would give SJW a reason to call out on our behavior.

**Comment by nukeop** [comment]
> Is that from our repo or their repo?

it's from open-letter

**Comment by jordigh** [comment]
You don't need to do much data mining. The sets are very obvious at a glance.

The original letter is full of people who have worked in free software over the years. Many of them are employed at companies that make proprietary software. This is absolutely obvious. You don't have to do any data mining, I see lots of friends I met at debconf, Pycon, people I have worked on bugs with. There's no need to come up with a conspiracy theory about how this letter is an insidious plot orchestrated by Microsoft and Google to destroy free software.

It's also absolutely obvious that the counterletter is being heavily promoted in Russia and 4chan, hence why almost all of the names in the counterletter are Russian. No need to datamine to figure that out either; you can see it at a glance.

> Is that from our repo or their repo?

@nukeop  It's this repo. I saw this data reported in 4chan's /g/, which is apparently where most of the counterletter is being organised. By the 4chan post, this data is just a simple grep and count of the list of signatories, since many of us are openly announcing our affiliations.

**Comment by imachug** [comment]
> the counterletter is being heavily promoted in Russia

Not really 'promoted', we just have a site that **every** Russian developer reads--[Habr](https://habr.com), and the letter was posted there. I think no other regions have a similar site, hence the bias.

**Comment by shenlebantongying** [comment]
I don't think the bias is that extreme. If you check one by one, there are ppl from all over the world, including friends from Asian, Australia and Africa. Believe it or not, there are free software activities on the third world too.

Besides, due to the pressure from "popular opinion", I think a lot of people actually cannot support us. They might lose job for signing our letter.

Even though a lot of them don't work for big companies, we are still users and developers for free software.

**Comment by nukeop** [comment]
If hackernews didn't nip the post about this letter in the bud, we'd be at 10k signatures right now easily

**Comment by jordigh** [comment]
> If hackernews didn't nip the post about this letter in the bud, we'd be at 10k signatures right now easily

Well, keep trying in 4chan, I'm sure you'll get there!

**Comment by M-i-k-o-t-o** [comment]
> It's also absolutely obvious that the counterletter is being heavily promoted in Russia and 4chan, hence why almost all of the names in the counterletter are Russian. No need to datamine to figure that out either; you can see it at a glance.

Uh oh, beware of the evil Russian hackers!!

Seriously... Please cool it with your racism.

**Comment by jordigh** [comment]
Hm?

There's nothing evil about Russians, but if you want to figure out who is signing each letter, you don't need to do anything but quickly look at the list of names. It's super-obvious. Or you can even do something as simple as what /g/ did and just grep the original letter. All of the affiliations are there. I don't know why the counterletter doesn't list affiliations, but almost all of the names are obviously Russian, at least initially. 

There has been some outreach outside Russia now, but the counterletter is still mostly Russia.

**Comment by RealJTG** [comment]
> > the counterletter is being heavily promoted in Russia
> 
> Not really 'promoted', we just have a site that **every** Russian developer reads--[Habr](https://habr.com), and the letter was posted there. I think no other region has a similar site, hence the bias.

There is quite popular Hacker News, but submission about RMS support letter got `[flagged]` tag for a some reason, so it gets a penalty and drops in the order, then it probably will disappear https://news.ycombinator.com/item?id=26565107

> the counterletter is being heavily promoted in Russia

I'd say it's rather counterletter is being heavily dis-promoted in US.

**Comment by imachug** [comment]
> There is quite popular Hacker News, but submission about RMS support letter got [flagged] tag for a some reason, so it gets a penalty and drops in the order and nobody actually could see it, then it probably will disappear https://news.ycombinator.com/item?id=26565107

Oh wow! I posted the link to the support letter on Hackernews too and got flagged too. What a coincidence!

**Comment by jordigh** [comment]
There's also no need to look for a conspiracy here. Hacker News stories are flagged by other users, not by the site admins. If enough people flag it, the story disappears, and it doesn't take that many flags.

So a few people in HN didn't want to deal with the flame war that this letter would produce. The original letter would probably also be equally flagged, just like the counterletter. HN stories often don't want to hear drama stories, particularly when they've already heard them once.

**Comment by shenlebantongying** [comment]
Some outsiders criticize that some of our signers' account is new or don't have much activity.

We have the same number of signers as them now. However, we have 3.1k PR and 2.7K forks, and the open letters only have 2.3 PR and 2K forks.

This may imply that a lot of their signers comes from email and added in bulk. Do those people don't even know how to create a PR or using git? That's interesting problem.

**Comment by jordigh** [comment]
We know, we just don't want to use the github interface for creating PRs. At least that was my reason for signing with email. It's a lot easier to send an email.

**Comment by imachug** [comment]
I stand for clearness. **All** signatures we receive have a public source: either it's a comment on a public issue, or a PR, or a email that we publish to #3105. rms-open-letter does not publish sources of bulk signatures.

**Comment by Alessandro-Barbieri** [comment]
Do you want a takedown because of violating GDPR?

**Comment by pizdjuk** [comment]
> Hm?
> 
> There's nothing evil about Russians, but if you want to figure out who is signing each letter, you don't need to do anything but quickly look at the list of names. It's super-obvious. Or you can even do something as simple as what /g/ did and just grep the original letter. Many of the affiliations are there. I don't know why the counterletter doesn't list affiliations, but almost all of the names are obviously Russian, at least initially.
> 
> There has been some outreach outside Russia now, but the counterletter is still mostly Russia.

not Russia, but russian speaking. There is a great difference.

**Comment by BlueFlo0d** [comment]
> I stand for clearness. **All** signatures we receive have a public source: either it's a comment on a public issue, or a PR, or a email that we publish to #3105. rms-open-letter does not publish sources of bulk signatures.

I'm maintaining a (best-effort) list of GitHub account signing the against letter here. https://github.com/BlueFlo0d/fashy-detector/blob/main/github-users.txt

**Comment by KOLANICH** [comment]
@BlueFlo0d, this list may be completely useless. Why? Because the only reliable info on the anti-RMS letter can be got only using GitHub. The rest of info mined from the list can be completely bullshit and misinformataion, and we shouldn't trust it and shouldn't relay it. In fact it contained quite some obvious trolling names of non-existent people, like [literally "a girl with penis"](https://github.com/rms-open-letter/rms-open-letter.github.io/commit/3decc7152f176f7efd495a70273e8a1fbb518670#diff-a48746cae70c44e7e105b594aad338ddd105c93c1cb445a40ba6aab785ba69e5) and [Vlad(imir)Len(in)a (yes some people were really named in favour of Lenin) daughter-of-sucking-(gerund grammatically)-of-dicks (it is patronim, not a surname, in Russia the respectful way to call a person (especially the one who is older or has a higher status in the situation) is &lt;name&gt; &lt;patronim&gt;)](https://github.com/rms-open-letter/rms-open-letter.github.io/commit/02983d978fff0815c7c99c3f5c353751359d4c01#diff-a48746cae70c44e7e105b594aad338ddd105c93c1cb445a40ba6aab785ba69e5L3106)

**Comment by BlueFlo0d** [comment]
> @BlueFlo0d, this list may be completely useless. Why? Because the only reliable info on the anti-RMS letter can be got only using GitHub. The rest of info mined from the list can be completely bullshit and misinformataion, and we shouldn't trust it and shouldn't relay it. In fact it contained quite some obvious (for the ones knowing Russian) trolling names of non-existent people, like [literally "a girl with penis" from "asshole-of-a-homosexulaist-labs"](https://github.com/rms-open-letter/rms-open-letter.github.io/commit/3decc7152f176f7efd495a70273e8a1fbb518670#diff-a48746cae70c44e7e105b594aad338ddd105c93c1cb445a40ba6aab785ba69e5) and [Vlad(imir)Len(in)a (yes some people were really named in favour of Lenin) daughter-of-sucking-(gerund grammatically)-of-dicks (it is patronim (though may be a matronim in extremily rare circumstances), not a surname, in Russia the respectful way to call a person (especially the one who is older or has a higher status in the situation) is <first name> <patronim>)](https://github.com/rms-open-letter/rms-open-letter.github.io/commit/02983d978fff0815c7c99c3f5c353751359d4c01#diff-a48746cae70c44e7e105b594aad338ddd105c93c1cb445a40ba6aab785ba69e5L3106) which at least shows that they haven't done basic checks when acdepting signatures, even if the anti-RMS letter maintainers recorded the signatures faithfully, if anyone can send there bullshit without any difficulty meant that the we cannot even estimate the amount of bullshit there, it may be 1 record, or it may be the whole base of signatories accepted via email.

Yes I'm aware of that, therefore I'm retrieving information only from commit history and repository metadata.
Currently no name is ever used.

The name maybe used to improve confident level of the mined GitHub IDs though. We know that name can be non sense, but if it's the same name as the GitHub account profile says (and if the GitHub account has sufficient activity), then it's a good indicator that we get a genuine entry.

**Comment by BlueFlo0d** [comment]
Also I only aim for a snapshot of the data before they closed GitHub PR and switch to email completely.
I agree that now-added new entries may contain little to no information.

**Comment by jordigh** [comment]
Why is the counterletter more reliable? All you have to do is create a Github account. That's pretty easy to do and easy to fake too.

And yeah, some mistakes happen, like someone who didn't know Russian didn't realise that the Russian name was a troll.

I mean, it's not a big surprise that Russia really hates the original letter, so of course Russia is trolling it and then feeling so happy that Russia managed to get mistakes in before they were noticed.

Good work, Russia. Do you want RMS? He can stay on your couch for two months and yell at you every time you accidentally say "Linux" instead of "GNU/Linux". You can have him.

**Comment by shenlebantongying** [comment]
Hint -> jordigh helped the translation of the open letter.

**Comment by jordigh** [comment]
4chan-level of detective skills there, chief. Watcha gonna do with your startling discovery?

**Comment by jordigh** [comment]
If you Russian just had to deal with RMS and actually knew him and actually spent time with him and if he actually went to more conferences in Russia, you might be sick of him too.

**Comment by nukeop** [comment]
Why does naming them always cause them to lash out?

**Comment by jordigh** [comment]
Lashing out? Like you guys are doing here? You're real happy to be in this extension too, aren't you?

https://github.com/aaronbassett/rms-letter-sigs/issues

**Comment by KOLANICH** [comment]
@nukeop, I don't think he said something deserving blocking. I think he should be unblocked.

@jordigh, 

>Why is the counterletter more reliable?

1. at least our transparency already makes it more reliable. One can check that the people have really sent PRs to us

2. >  All you have to do is create a Github account.

IMHO only votes of people with FOSS contributions should be counted. Making a PR that is merged into real projects is not easy. Making more than 1 merged PR is even more difficult.

>That's pretty easy to do and easy to fake too.

Not very easy. Even registering a email without telling it a phone number is not easy in 2021.

>And yeah, some mistakes happen, like someone who didn't know Russian didn't realise that the Russian name was a troll.

It is irrelevant. The name just makes it obvious that anti-RMS letter accepts "signatures" without enough checks. This automatically means that whole signatures accepted this way are not valid. At  least until, the contrary is proved.

>I mean, it's not a big surprise that Russia really hates the original letter

You shouldn't tell on behalf of whole Russia. Most of population of Russian Federation and Russia just don't give a f**k about what is happening with Stallman. They don't even know who Stallman is.

>so of course Russia is trolling it and then feeling so happy that Russia managed to get mistakes in before they were noticed.

It shows that the anti-rms letter doesn't have sufficient checks. If opponents of the anti-rms letter have managed to send some obviously and explicitly fake signatures without being noticed, how many ones the proponents of RMS removal directly interested in making the results be as high as possible could have sent?

**Comment by nukeop** [comment]
> @nukeop, I don't think he said something deserving blocking. I think he should be unblocked.

He only came here to troll, not participate in the discussion in a meaningful way, so it's just noise.

**Comment by 6r1d** [comment]
@KOLANICH, while I agree with your arguments, I don't think it's time to talk to people who come and shout their point of view in our ears. I already had some spam in many places, was told to avoid some, etc.
If someone comes, listens to our arguments and thinks, it's one thing, but that's what generally leads to people helping us or staying neutral, does it not?

**Comment by KOLANICH** [comment]
Even if he has anti-stallman point, it doesn't me that he should be blocked.
